src/Ledger/Ratify.lagda

```diff
@@ -249,22 +254,18 @@ module _
         (no _) → constMap (dom (cc ˢ)) Vote.no
       nothing → ∅ᵐ
 
-  actualPDRepVotes : VDeleg ⇀ Vote
-  actualPDRepVotes = ❴ abstainRep      , Vote.abstain ❵ᵐ
-                 ∪ᵐˡ ❴ noConfidenceRep , (case ga of λ where
-                                           NoConfidence → Vote.yes
-                                           _            → Vote.no) ❵ᵐ
-  actualDRepVotes : VDeleg ⇀ Vote
-  actualDRepVotes = mapKeys (uncurry credVoter) (roleVotes GovRole.DRep) (λ where _ _ refl → refl)
-                  ∪ᵐˡ constMap (map (credVoter DRep) activeDReps) Vote.no
-    where
-      activeDReps : ℙ Credential
-      activeDReps = dom (filterᵐ (to-sp (currentEpoch ≤ᵉ?_ ∘ proj₂)) dreps ˢ)
-
-  actualVotes : VDeleg ⇀ Vote
-  actualVotes = mapKeys (credVoter CC) actualCCVotes (λ where _ _ refl → refl)
-              ∪ᵐˡ actualPDRepVotes ∪ᵐˡ actualDRepVotes
-              ∪ᵐˡ mapKeys (uncurry credVoter) votes (λ where _ _ refl → refl)
+  actualPDRepVotes actualDRepVotes actualVotes : VDeleg ⇀ Vote
+  actualPDRepVotes  =    ❴ abstainRep       , Vote.abstain ❵ᵐ
+                    ∪ᵐˡ  ❴ noConfidenceRep  , (case ga of λ where
+                                                NoConfidence  → Vote.yes
+                                                _             → Vote.no) ❵ᵐ
+  actualDRepVotes =
+    let activeDReps = dom (filterᵐ (to-sp (currentEpoch ≤ᵉ?_ ∘ proj₂)) dreps ˢ) in
```

Comment:
I think `where` is much more readable in this case

Reply:
Yeah, I was thinking with my type-checking hat on here, since `where` introduces a whole module but we do not need that here. It is indeed unfortunate that Agda does not have equivalent let/where semantics, we would want to use them interchangeably depending on the most readable option.

I would argue that we should follow this generally for **hidden code**, and freely use `where` for PDF readability without care for type-checking/internals.

Reply:
I don't think we should worry about internals here (but [this](https://github.com/agda/agda-categories/wiki/speed) suggests `where` is better for performance). This also needs to go into the PDF, so readability is more important anyway.